components/kyma-environment-broker/internal/model.go

```diff
@@ -138,6 +138,84 @@ type Orchestration struct {
 	RuntimeOperations sql.NullString
 }
 
+const (
+	Pending    = "pending"
+	InProgress = "in progress"
+	Succeeded  = "succeeded"
+	Failed     = "failed"
+)
+
+// Runtime is the data type which captures the needed SKR specific attributes to perform reconciliations on a given runtime.
+type Runtime struct {
+	InstanceID      string `json:"instanceId"`
+	RuntimeID       string `json:"runtimeId"`
+	GlobalAccountID string `json:"globalAccountId"`
+	SubAccountID    string `json:"subaccountId"`
+	// The corresponding shoot cluster's .metadata.name value
+	ShootName string `json:"shootName"`
+	// The corresponding shoot cluster's .spec.maintenance.timeWindow.Begin value, which is in in "HHMMSS+[HHMM TZ]" format, e.g. "040000+0000"
+	MaintenanceWindowBegin time.Time `json:"maintenanceWindowBegin"`
+	// The corresponding shoot cluster's .spec.maintenance.timeWindow.End value, which is in "HHMMSS+[HHMM TZ]" format, e.g. "040000+0000"
+	MaintenanceWindowEnd time.Time `json:"maintenanceWindowEnd"`
+}
+
+// RuntimeOperation encapsulates a Runtime object and an operation ID for the OrchestrationStrategy to execute.
+type RuntimeOperation struct {
+	Runtime
+	OperationID string `json:"operationId"`
+	Status      string `json:"status,omitempty"`
+}
+
+// TargetAll all SKRs provisioned successfully and not deprovisioning
+const TargetAll = "all"
+
+// RuntimeTarget captures a specification of SKR targets to resolve for an orchestration.
+// When a RuntimeTarget defines multiple fields, all should match to any given runtime to be selected (i.e. the terms are AND-ed).
+type RuntimeTarget struct {
+	// Valid values: "all"
+	Target string `json:"target,omitempty"`
+	// Regex pattern to match against the runtime's GlobalAccount field. E.g. CA50125541TID000000000741207136, CA.*
+	GlobalAccount string `json:"globalAccount,omitempty"`
+	// Regex pattern to match against the runtime's SubAccount field. E.g. 0d20e315-d0b4-48a2-9512-49bc8eb03cd1
+	SubAccount string `json:"subAccount,omitempty"`
+	// Regex pattern to match against the shoot cluster's Region field (not SCP platform-region). E.g. "europe|eu-"
+	Region string `json:"region,omitempty"`
+	// RuntimeID is used to indicate a specific runtime
+	RuntimeID string `json:"runtimeId,omitempty"`
+}
+
+type StrategyType string
+
+const (
+	ParallelStrategy StrategyType = "parallel"
```

Comment:
those are not used at the moment, consider commenting it or removing to avoid creating dead code